### PR TITLE
Votes should be hashed according to EIP-712 #335

### PR DESCRIPTION
**Pull request by deepesh-kn**
Fixes: #335
- Created `struct VoteMessage`
- Created `struct Vote`
- Added Hashing logic as per EIP-712

Please note:  In this  PR `bytes20 coreIdentifier` is still part of `struct VoteMessage`. 
As the votes are for both origin and auxiliary chain, we need to have the `coreIdentifier` information while generating the `voteHash`.

### CODE REVIEW

File: contracts/core/PollingPlace.sol [hunk 1]
```diff
@@ -300,28 +346,27 @@ contract PollingPlace is PollingPlaceInterface {
...
-            _targetHeight
-        );
-        Validator storage validator = getValidatorFromVote(
-            voteHash,
+            _targetHeight,
             _v,
             _r,
-            _s
-        );
+            _s);
```
Comment (schemar) [inline contracts/core/PollingPlace.sol]:
```suggestion
            _s
        );
```

Comment (deepesh-kn) [reply contracts/core/PollingPlace.sol]: Done!

**Comment by deepesh-kn** [comment]
`Did you check whether it is possible to calculate the vote hash only once?`
      Yes, that was missed in the initial commit, I committed that part now, thanks!!!

File: contracts/core/PollingPlace.sol [hunk 2]
```diff
@@ -70,6 +70,52 @@ contract PollingPlace is PollingPlaceInterface {
         uint256 endHeight;
     }
 
+    /** Vote message */
+    struct VoteMessage {
+
+        /** A unique identifier that identifies what chain this vote is about. */
```
Comment (schemar) [inline contracts/core/PollingPlace.sol]: Please also explain here why it has to be part of the struct.

**Pull request merged**